### DISCUSSION TITLE
Consider renaming `shift`

### DISCUSSION THREAD

**Discussion by aleclarson** [radashi-org/radashi#103] [Ideas]
Since there's already `Array.prototype.shift` and both work with arrays, it's better to avoid reusing the name “shift”.

**What does Radashi `shift` do?**  
When `n` is positive, it moves `n` items from the end of an array to the front (preserving order). When `n` is negative, it moves the front `n` items of an array to the end.

**Possible alternatives**  

- rotate (too vague in my opinion)
- revolve (sounds the coolest!)
- circShift (used by [Matlab](https://fr.mathworks.com/help/matlab/ref/circshift.html))
- circularShift
- circularMove
- displace
- wrap

**Should it be removed entirely?**  
It's unclear how useful this function is, or if it deserves to be in Radashi. Any feedback on this is appreciated.

**Comment by adamhamlin** [comment]
I've seen this called `rotate` in other contexts, but I agree the semantics of that aren't intuitive. I think `circularShift` is probably the most intuitive on your list.

But yeah barring some like highly-specific computer vision use case or similar, seems like more a leet code problem than anything. I could see maybe a method like `getNextElement(arr, curr)` that returns the next element in the list and handles wrapping around. But using `circularShift` to do that wouldn't be very efficient 🤷

**Comment by eumkz** [comment]
I think `circularShift` is a good name, but I'm not sure this function is essential. It seems pretty niche. Unless there's a common use case I'm missing, it might be better to remove it and keep the library focused on more widely-used operations.

**Comment by nnmrts** [comment]
I disagree that this is not an essential function. My recent personal use-case was iterating over polygon vertices from any starting vertex. `shift` enabled me to do that very elegantly.

But there are many more valid use-cases:

- https://stackoverflow.com/questions/19899797/applications-of-a-circular-shift
- Image processing
- String manipulation
- Game development

However I agree it should be renamed to `circularShift`.

**Comment by crishoj** [comment]
`circularShift` seems very clear.

**Comment by aleclarson** [comment]
Resolved by #156
